test/NuGet.Core.FuncTests/NuGet.Packaging.FuncTest/SigningTests/TimestampProviderTests.cs

```diff
@@ -331,14 +340,104 @@ public TimestampProviderTests(SigningTestFixture fixture)
+                foreach (var attr in primarySignature.SignerInfo.UnsignedAttributes)
+                {
+                    if (string.Compare(attr.Oid.Value, Oids.SignatureTimeStampTokenAttribute, CultureInfo.CurrentCulture, CompareOptions.Ordinal) == 0)
+                    {
+                        hasTimestampUnsignedAttribute = true;
+                    }
+                }
+                hasTimestampUnsignedAttribute.Should().BeTrue();
+            }
+        }
+
+        [CIOnlyFact]
+        public async Task TimestampSignatureAsync_TimestampingCountersignature_Succeds()
+        {
+            var logger = new TestLogger();
+            var timestampService = await _testFixture.GetDefaultTrustedTimestampServiceAsync();
+            var timestampProvider = new Rfc3161TimestampProvider(timestampService.Url);
+            var signatureContent = new SignatureContent(SigningSpecifications.V1, Common.HashAlgorithmName.SHA256, "Test data to be signed and timestamped");
+
+            using (var authorCert = new X509Certificate2(_trustedTestCert.Source.Cert))
+            {
+                var timestampHashAlgorithm = Common.HashAlgorithmName.SHA256;
+                var signedCms = SigningTestUtility.GenerateRepositoryCountersignedSignedCms(authorCert, signatureContent.GetBytes());
+                var signature = PrimarySignature.Load(signedCms.Encode());
+                var signatureValue = signature.GetSignatureValue();
+                var messageHash = timestampHashAlgorithm.ComputeHash(signatureValue);
+
+                var request = new TimestampRequest(
+                   signingSpec: SigningSpecifications.V1,
+                   signatureMessageHash: messageHash,
+                   timestampHashAlgorithm: timestampHashAlgorithm,
+                   timestampSignaturePlacement: SignaturePlacement.Countersignature
+               );
+
+                // Act
+                var primarySignature = await timestampProvider.TimestampSignatureAsync(signature, request, logger, CancellationToken.None);
+
+                var repositoryCountersignature = RepositoryCountersignature.GetRepositoryCountersignature(primarySignature);
+
+                // Assert
+                repositoryCountersignature.Should().NotBeNull();
+                repositoryCountersignature.SignerInfo.Should().NotBeNull();
+                repositoryCountersignature.SignerInfo.UnsignedAttributes.Count.Should().BeGreaterOrEqualTo(1);
+
+                var hasTimestampUnsignedAttribute = false;
+                foreach (var attr in repositoryCountersignature.SignerInfo.UnsignedAttributes)
+                {
+                    if (string.Compare(attr.Oid.Value, Oids.SignatureTimeStampTokenAttribute, CultureInfo.CurrentCulture, CompareOptions.Ordinal) == 0)
+                    {
+                        hasTimestampUnsignedAttribute = true;
+                    }
+                }
+                hasTimestampUnsignedAttribute.Should().BeTrue();
```

Comment:
Also verify the attribute value.  Decode the signed CMS and call `SignedCms.CheckSignature(verifySignatureOnly: true)`.